broker-daemon/block-order-worker/index.js

```diff
@@ -316,6 +281,43 @@ class BlockOrderWorker extends EventEmitter {
     }
   }
 
+  /**
+   * Move a block order to a completed state if all orders have been completed
+   * @todo What should we do if this method fails, but the order itself is completed?
+   * @param {String} blockOrderId
+   */
+  async completeBlockOrder (blockOrderId) {
+    this.logger.info('Attempting to put block order in a completed state', { id: blockOrderId })
+
+    const blockOrder = await this.getBlockOrder(blockOrderId)
+
+    // check the fillamount on each collection of state machines from the block order
+    // and make sure that either is equal to how much we are trying to fill.
+    let totalFilled = Big(0)
+    totalFilled = blockOrder.fills.reduce((acc, fsm) => acc.plus(fsm.fill.fillAmount), totalFilled)
+    totalFilled = blockOrder.openOrders.reduce((acc, osm) => {
+      // If the order has not been filled yet, then the `fillAmount` will be undefined
+      // so we instead default to 0
+      return acc.plus(osm.order.fillAmount || 0)
+    }, totalFilled)
+
+    this.logger.debug('Current total filled amount: ', { totalFilled, blockOrderAmount: blockOrder.baseAmount })
+
+    const stillBeingFilled = totalFilled.lt(blockOrder.baseAmount)
+    const activeOrders = blockOrder.openOrders.some(osm => osm.order.active)
+    const activeFills = blockOrder.fills.some(fsm => fsm.fill.active)
+
+    if (!activeOrders && !activeFills && !stillBeingFilled) {
```

Comment:
if it is not still being filled and it has active orders, what behavior do we want?

Reply:
@treygriffith we intentionally removed checks for that functionality